// @ariakit/core/cjs/__chunks/HOEZKRUM.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";


var _MYTMMWXIcjs = require('./MYTMMWXI.cjs');





var _FJSZTIOUcjs = require('./FJSZTIOU.cjs');


var _3UT5FE6Kcjs = require('./3UT5FE6K.cjs');




var _AV6KTKLEcjs = require('./AV6KTKLE.cjs');

// src/popover/popover-store.ts
function createPopoverStore(_a = {}) {
  var _b = _a, {
    popover: otherPopover
  } = _b, props = _AV6KTKLEcjs.__objRest.call(void 0, _b, [
    "popover"
  ]);
  const store = _FJSZTIOUcjs.mergeStore.call(void 0, 
    props.store,
    _FJSZTIOUcjs.omit.call(void 0, otherPopover, [
      "arrowElement",
      "anchorElement",
      "contentElement",
      "popoverElement",
      "disclosureElement"
    ])
  );
  _FJSZTIOUcjs.throwOnConflictingProps.call(void 0, props, store);
  const syncState = store == null ? void 0 : store.getState();
  const dialog = _MYTMMWXIcjs.createDialogStore.call(void 0, _AV6KTKLEcjs.__spreadProps.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, {}, props), { store }));
  const placement = _3UT5FE6Kcjs.defaultValue.call(void 0, 
    props.placement,
    syncState == null ? void 0 : syncState.placement,
    "bottom"
  );
  const initialState = _AV6KTKLEcjs.__spreadProps.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, {}, dialog.getState()), {
    placement,
    currentPlacement: placement,
    anchorElement: _3UT5FE6Kcjs.defaultValue.call(void 0, syncState == null ? void 0 : syncState.anchorElement, null),
    popoverElement: _3UT5FE6Kcjs.defaultValue.call(void 0, syncState == null ? void 0 : syncState.popoverElement, null),
    arrowElement: _3UT5FE6Kcjs.defaultValue.call(void 0, syncState == null ? void 0 : syncState.arrowElement, null),
    rendered: Symbol("rendered")
  });
  const popover = _FJSZTIOUcjs.createStore.call(void 0, initialState, dialog, store);
  return _AV6KTKLEcjs.__spreadProps.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, {}, dialog), popover), {
    setAnchorElement: (element) => popover.setState("anchorElement", element),
    setPopoverElement: (element) => popover.setState("popoverElement", element),
    setArrowElement: (element) => popover.setState("arrowElement", element),
    render: () => popover.setState("rendered", Symbol("rendered"))
  });
}



exports.createPopoverStore = createPopoverStore;
