// @ariakit/core/cjs/menu/menu-store.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";


var _XVR5QRC3cjs = require('../__chunks/XVR5QRC3.cjs');
require('../__chunks/HOEZKRUM.cjs');
require('../__chunks/MYTMMWXI.cjs');
require('../__chunks/F6Q5QSXF.cjs');


var _AR2VJ37Fcjs = require('../__chunks/AR2VJ37F.cjs');
require('../__chunks/FFY2GSOG.cjs');








var _FJSZTIOUcjs = require('../__chunks/FJSZTIOU.cjs');



var _3UT5FE6Kcjs = require('../__chunks/3UT5FE6K.cjs');
require('../__chunks/FTB5CXVB.cjs');
require('../__chunks/ULSPM3Y3.cjs');




var _AV6KTKLEcjs = require('../__chunks/AV6KTKLE.cjs');

// src/menu/menu-store.ts
function createMenuStore(_a = {}) {
  var _b = _a, {
    combobox,
    parent,
    menubar
  } = _b, props = _AV6KTKLEcjs.__objRest.call(void 0, _b, [
    "combobox",
    "parent",
    "menubar"
  ]);
  const parentIsMenubar = !!menubar && !parent;
  const store = _FJSZTIOUcjs.mergeStore.call(void 0, 
    props.store,
    _FJSZTIOUcjs.pick.call(void 0, parent, ["values"]),
    _FJSZTIOUcjs.omit.call(void 0, combobox, [
      "arrowElement",
      "anchorElement",
      "contentElement",
      "popoverElement",
      "disclosureElement"
    ])
  );
  _FJSZTIOUcjs.throwOnConflictingProps.call(void 0, props, store);
  const syncState = store.getState();
  const composite = _AR2VJ37Fcjs.createCompositeStore.call(void 0, _AV6KTKLEcjs.__spreadProps.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, {}, props), {
    store,
    orientation: _3UT5FE6Kcjs.defaultValue.call(void 0, 
      props.orientation,
      syncState.orientation,
      "vertical"
    )
  }));
  const hovercard = _XVR5QRC3cjs.createHovercardStore.call(void 0, _AV6KTKLEcjs.__spreadProps.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, {}, props), {
    store,
    placement: _3UT5FE6Kcjs.defaultValue.call(void 0, 
      props.placement,
      syncState.placement,
      "bottom-start"
    ),
    timeout: _3UT5FE6Kcjs.defaultValue.call(void 0, 
      props.timeout,
      syncState.timeout,
      parentIsMenubar ? 0 : 150
    ),
    hideTimeout: _3UT5FE6Kcjs.defaultValue.call(void 0, props.hideTimeout, syncState.hideTimeout, 0)
  }));
  const initialState = _AV6KTKLEcjs.__spreadProps.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, {}, composite.getState()), hovercard.getState()), {
    initialFocus: _3UT5FE6Kcjs.defaultValue.call(void 0, syncState.initialFocus, "container"),
    values: _3UT5FE6Kcjs.defaultValue.call(void 0, 
      props.values,
      syncState.values,
      props.defaultValues,
      {}
    )
  });
  const menu = _FJSZTIOUcjs.createStore.call(void 0, initialState, composite, hovercard, store);
  _FJSZTIOUcjs.setup.call(void 0, 
    menu,
    () => _FJSZTIOUcjs.sync.call(void 0, menu, ["mounted"], (state) => {
      if (state.mounted)
        return;
      menu.setState("activeId", null);
    })
  );
  _FJSZTIOUcjs.setup.call(void 0, 
    menu,
    () => _FJSZTIOUcjs.sync.call(void 0, parent, ["orientation"], (state) => {
      menu.setState(
        "placement",
        state.orientation === "vertical" ? "right-start" : "bottom-start"
      );
    })
  );
  return _AV6KTKLEcjs.__spreadProps.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, {}, composite), hovercard), menu), {
    combobox,
    parent,
    menubar,
    hideAll: () => {
      hovercard.hide();
      parent == null ? void 0 : parent.hideAll();
    },
    setInitialFocus: (value) => menu.setState("initialFocus", value),
    setValues: (values) => menu.setState("values", values),
    setValue: (name, value) => {
      if (name === "__proto__")
        return;
      if (name === "constructor")
        return;
      if (Array.isArray(name))
        return;
      menu.setState("values", (values) => {
        const prevValue = values[name];
        const nextValue = _3UT5FE6Kcjs.applyState.call(void 0, value, prevValue);
        if (nextValue === prevValue)
          return values;
        return _AV6KTKLEcjs.__spreadProps.call(void 0, _AV6KTKLEcjs.__spreadValues.call(void 0, {}, values), {
          [name]: nextValue !== void 0 && nextValue
        });
      });
    }
  });
}


exports.createMenuStore = createMenuStore;
